// office-ui-fabric-react/lib-amd/demo/pages/Facepile/examples/Facepile.Basic.Example.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
define(["require", "exports", 'react', '../../../../index', './FacepileExampleData', './Facepile.Examples.scss'], function (require, exports, React, index_1, FacepileExampleData_1) {
    "use strict";
    var FacepileBasicExample = (function (_super) {
        __extends(FacepileBasicExample, _super);
        function FacepileBasicExample() {
            _super.call(this);
            this.state = {
                numberOfFaces: 1
            };
        }
        FacepileBasicExample.prototype.render = function () {
            var _this = this;
            var numberOfFaces = this.state.numberOfFaces;
            var facepileProps = {
                personas: FacepileExampleData_1.facepilePersonas.slice(0, numberOfFaces),
                chevronButtonProps: {
                    onClick: function (ev) {
                        return alert('Down arrow icon clicked');
                    }
                }
            };
            return (React.createElement("div", {className: 'ms-FacepileExample'}, 
                React.createElement(index_1.Facepile, __assign({}, facepileProps)), 
                React.createElement(index_1.Slider, {label: 'Number of Personas:', min: 1, max: 8, step: 1, showValue: true, value: this.state.numberOfFaces, onChange: function (value) { return _this.setState({ numberOfFaces: value }); }})));
        };
        return FacepileBasicExample;
    }(React.Component));
    exports.FacepileBasicExample = FacepileBasicExample;
});
