// office-ui-fabric-react/lib-amd/demo/pages/Facepile/examples/Facepile.Overflow.Example.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
define(["require", "exports", 'react', '../../../../index', './FacepileExampleData', './Facepile.Examples.scss'], function (require, exports, React, index_1, FacepileExampleData_1) {
    "use strict";
    var facepileProps = {
        personas: FacepileExampleData_1.facepilePersonas,
        maxDisplayablePersonas: 5,
        overflowButtonProps: {
            onClick: function (ev) {
                return alert('overflow icon clicked');
            }
        }
    };
    var FacepileOverflowExample = (function (_super) {
        __extends(FacepileOverflowExample, _super);
        function FacepileOverflowExample() {
            _super.call(this);
            this.state = {
                displayedPersonas: 5
            };
        }
        FacepileOverflowExample.prototype.render = function () {
            var _this = this;
            var displayedPersonas = this.state.displayedPersonas;
            facepileProps.maxDisplayablePersonas = displayedPersonas;
            return (React.createElement("div", {className: 'ms-FacepileExample'}, 
                React.createElement(index_1.Facepile, __assign({}, facepileProps)), 
                React.createElement(index_1.Slider, {label: 'Number of Personas Shown:', min: 0, max: 6, step: 1, showValue: true, value: this.state.displayedPersonas, onChange: function (value) { return _this.setState({ displayedPersonas: value }); }})));
        };
        return FacepileOverflowExample;
    }(React.Component));
    exports.FacepileOverflowExample = FacepileOverflowExample;
});
